audioipc/src/ipccore.rs

```diff
@@ -356,28 +356,51 @@ impl Connection {
     // Connections are always interested in READABLE.  clear_readable is only
     // called when the connection is in the process of shutting down.
     fn clear_readable(&mut self, registry: &Registry) -> Result<()> {
-        self.interest.and_then(|i| i.remove(Interest::READABLE));
-        self.update_registration(registry)
```

Comment:
It seems this also correct the `interest` value? The returned value from `self.interest.and_then(|i| i.remove(Interest::READABLE))` was not set to `update_registration`.

Reply:
Yeah; I thought I had already fixed that already but it must've got lost on my dev branch.